ignite/handlers/state_param_scheduler.py

```diff
@@ -0,0 +1,374 @@
+        User defined function is taking an event index as input and returns parameter value.
+
+    Args:
+        lambda_fn: user defined parameter update function.
+        param_name: name of parameter to update.
+        save_history: whether to log the parameter values to
+            `engine.state.param_history`, (default=False).
+
+    Examples:
+
+        .. code-block:: python
+
+            ...
+            engine = Engine(train_step)
+
+            initial_value = 10
+            gamma = 0.99
+
+            lambda_scheduler = LambdaStateScheduler(
+                param_name="lambda",
+                lambda_fn=lambda event_index: initial_value * gamma ** (event_index % 9),
+            )
+
+            lambda_state_parameter_scheduler.attach(engine, Events.EPOCH_COMPLETED)
+
+            # basic handler to print scheduled state parameter
+            engine.add_event_handler(Events.EPOCH_COMPLETED, lambda _ : print(engine.state.custom_scheduled_param))
+
+            engine.run([0] * 8, max_epochs=2)
+
+    .. versionadded:: 0.6.0
+
+    """
+
+    def __init__(self, lambda_fn: Callable, param_name: str, save_history: bool = False):
+        super(LambdaStateScheduler, self).__init__(param_name, save_history)
+        self.lambda_fn = lambda_fn
+        self._state_attrs += ["lambda_fn"]
+
+    def get_param(self) -> Union[List[float], float]:
+        return self.lambda_fn(self.event_index)
+
+
+class PiecewiseLinearStateScheduler(StateParamScheduler):
+    """Piecewise linear state parameter scheduler.
+
+    Args:
+        milestones_values: list of tuples (event index, parameter value)
+            represents milestones and parameter values. Milestones should be increasing integers.
+        param_name: name of parameter to update.
+        save_history: whether to log the parameter values to
+            `engine.state.param_history`, (default=False).
+
+    Examples:
+
+        .. code-block:: python
+
+            ...
+            engine = Engine(train_step)
+
+            param_scheduler = PiecewiseLinearStateScheduler(
+                param_name="pw_linear_scheduled_param",
+                milestones_values=[(10, 0.5), (20, 0.45), (21, 0.3), (30, 0.1), (40, 0.1)]
+            )
+
+            pwlinear_state_parameter_scheduler.attach(engine, Events.EPOCH_COMPLETED)
+
+            # basic handler to print scheduled state parameter
+            engine.add_event_handler(Events.EPOCH_COMPLETED, lambda _ : print(engine.state.pw_linear_scheduled_param))
+
+            engine.run([0] * 8, max_epochs=40)
+            #
+            # Sets the state parameter value to 0.5 over the first 10 iterations, then decreases linearly from 0.5 to
+            # 0.45 between 10th and 20th iterations. Next there is a jump to 0.3 at the 21st iteration and the state
+            # parameter value decreases linearly from 0.3 to 0.1 between 21st and 30th iterations and remains 0.1 until
+            # the end of the iterations.
+            #
```

Comment:
@fco-dv please fix the docstring: reduce var names and ensure that the example can be executed.

Reply:
Indeed I will re-check examples and var names...